near-contract-standards/src/event.rs

```diff
@@ -51,64 +35,52 @@ pub enum Nep171EventKind<'a> {
 #[derive(Serialize, Deserialize, Debug)]
 pub struct NftMintData<'a> {
     #[serde(borrow)]
-    pub owner_id: Cow<'a, str>,
+    pub owner_id: &'a str,
     #[serde(borrow)]
-    pub token_ids: Vec<Cow<'a, str>>,
+    pub token_ids: Vec<&'a str>,
     #[serde(borrow)]
-    pub memo: Option<Cow<'a, str>>,
+    pub memo: Option<&'a str>,
 }
 
 impl<'a> NftMintData<'a> {
-    pub fn new<O, T, M>(owner_id: O, token_ids: Vec<T>, memo: Option<M>) -> NftMintData<'a>
-    where
-        O: Into<Cow<'a, str>>,
-        T: Into<Cow<'a, str>>,
-        M: Into<Cow<'a, str>>,
-    {
-        Self {
-            owner_id: owner_id.into(),
-            token_ids: token_ids.into_iter().map(Into::into).collect(),
-            memo: memo.map(Into::into),
-        }
+    pub fn new(
+        owner_id: &'a AccountId,
+        token_ids: Vec<&'a str>,
+        memo: Option<&'a str>,
+    ) -> NftMintData<'a> {
+        Self { owner_id: owner_id.as_str(), token_ids, memo }
     }
 }
 
 #[skip_serializing_none]
 #[derive(Serialize, Deserialize, Debug)]
 pub struct NftTransferData<'a> {
     #[serde(borrow)]
-    pub old_owner_id: Cow<'a, str>,
+    pub old_owner_id: &'a str,
     #[serde(borrow)]
-    pub new_owner_id: Cow<'a, str>,
+    pub new_owner_id: &'a str,
     #[serde(borrow)]
-    pub token_ids: Vec<Cow<'a, str>>,
+    pub token_ids: Vec<&'a str>,
     #[serde(borrow)]
-    pub authorized_id: Option<Cow<'a, str>>,
+    pub authorized_id: Option<&'a str>,
     #[serde(borrow)]
-    pub memo: Option<Cow<'a, str>>,
+    pub memo: Option<&'a str>,
 }
 
 impl<'a> NftTransferData<'a> {
-    pub fn new<O, N, T, A, M>(
-        old_owner_id: O,
-        new_owner_id: N,
-        token_ids: Vec<T>,
-        authorized_id: Option<A>,
-        memo: Option<M>,
-    ) -> NftTransferData<'a>
-    where
-        O: Into<Cow<'a, str>>,
-        N: Into<Cow<'a, str>>,
-        T: Into<Cow<'a, str>>,
-        A: Into<Cow<'a, str>>,
-        M: Into<Cow<'a, str>>,
-    {
+    pub fn new(
+        old_owner_id: &'a AccountId,
+        new_owner_id: &'a AccountId,
+        token_ids: Vec<&'a str>,
```

Comment:
Why are these not `AccountId`s?

Reply:
`token_id`'s?

Reply:
oh I didn't realize they are strings in internal impl. Can disregard this as it wouldn't come with this change